pyxlsb/styles.py

```diff
@@ -25,14 +63,48 @@ def _parse(self):
         self._cell_styles = list()
         self._cell_style_xfs = list()
 
+        self._xf_record = dict()
+        self._format_record = dict()
+
         self._fp.seek(0, os.SEEK_SET)
         for rectype, rec in RecordReader(self._fp):
             # TODO
-            if rectype == rt.END_STYLE_SHEET:
+            if rectype == 47:
+                self._xf_record[len(self._xf_record) - 1] = rec
```

Comment:
Use the `count` from the `BEGIN_CELL_XFS` record to expand the list. You can then simply assign to the proper index afterwards instead of growing the for each record.

MS-XLSB guarantees that XF records will fallow sequentially after a `BEGIN_CELL_XFS`.